File: app/app/src/main/java/com/example/impactx/data/local/SessionEntity.java
package com.example.impactx.data.local;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.Ignore;

@Entity(tableName = "user_session")
public class SessionEntity {
    @PrimaryKey
    @NonNull
    public String userId;
    
    public String username;
    public String correo;
    public String planActivo;
    public String accessToken;
    public String refreshToken;
    public long expiresAt;

    @Ignore
    public SessionEntity() {
        this.userId = "";
    }

    public SessionEntity(@NonNull String userId, String username, String correo, String planActivo, String accessToken, String refreshToken, long expiresAt) {
        this.userId = userId;
        this.username = username;
        this.correo = correo;
        this.planActivo = planActivo;
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.expiresAt = expiresAt;
    }
}

File: app/app/src/main/java/com/example/impactx/data/local/SessionEntity.kt
package com.example.impactx.data.local

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "user_session")
data class SessionEntity(
    @PrimaryKey val userId: String,
    val username: String,
    val correo: String,
    val planActivo: String?,
    val accessToken: String,
    val refreshToken: String,
    val expiresAt: Long // Timestamp in milliseconds when access token expires
)
